crates/neon/src/types_impl/bigint.rs

```diff
@@ -0,0 +1,377 @@
+        self.0
+    }
+}
+
+impl<T> fmt::Display for Error<T>
+where
+    T: fmt::Display,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Loss of precision reading BigInt ({})", self.0)
+    }
+}
+
+impl<T> error::Error for Error<T> where T: fmt::Display + fmt::Debug {}
+
+impl<T, E> ResultExt<T> for Result<T, Error<E>>
+where
+    E: fmt::Display,
+{
+    fn or_throw<'a, C: Context<'a>>(self, cx: &mut C) -> NeonResult<T> {
+        self.or_else(|err| cx.throw_range_error(err.to_string()))
+    }
+}
+
+impl JsBigInt {
+    pub const POSITIVE: Sign = Sign::Positive;
+    pub const NEGATIVE: Sign = Sign::Negative;
+
+    /// Creates a `BigInt` from an [`i64`].
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use neon::{prelude::*, types::JsBigInt};
+    /// # fn example(mut cx: FunctionContext) -> JsResult<JsBigInt> {
+    /// let value: Handle<JsBigInt> = JsBigInt::from_i64(&mut cx, 42);
+    /// # Ok(value)
+    /// # }
+    /// ```
+    pub fn from_i64<'cx, C>(cx: &mut C, n: i64) -> Handle<'cx, Self>
+    where
+        C: Context<'cx>,
+    {
+        let mut v = MaybeUninit::uninit();
+        let v = unsafe {
+            assert_eq!(
+                sys::create_bigint_int64(cx.env().to_raw(), n, v.as_mut_ptr(),),
+                sys::Status::Ok,
+            );
+
+            v.assume_init()
+        };
+
+        Handle::new_internal(Self(v))
+    }
+
+    /// Creates a `BigInt` from a [`u64`].
+    pub fn from_u64<'cx, C>(cx: &mut C, n: u64) -> Handle<'cx, Self>
+    where
+        C: Context<'cx>,
+    {
+        let mut v = MaybeUninit::uninit();
+        let v = unsafe {
+            assert_eq!(
+                sys::create_bigint_uint64(cx.env().to_raw(), n, v.as_mut_ptr(),),
+                sys::Status::Ok,
+            );
+
+            v.assume_init()
+        };
+
+        Handle::new_internal(Self(v))
+    }
+
+    // Internal helper for creating a _signed_ `BigInt` from a [`u128`] magnitude
+    fn from_u128_sign<'cx, C>(cx: &mut C, sign: Sign, n: u128) -> Handle<'cx, Self>
+    where
+        C: Context<'cx>,
+    {
+        let n = n.to_le();
+        let digits = [n as u64, (n >> 64) as u64];
+
+        Self::from_digits_le(cx, sign, &digits)
+    }
+
+    /// Creates a `BigInt` from an [`i128`].
+    pub fn from_i128<'cx, C>(cx: &mut C, n: i128) -> Handle<'cx, Self>
+    where
+        C: Context<'cx>,
+    {
+        if n >= 0 {
+            return Self::from_u128(cx, n as u128);
+        }
+
+        // Get the magnitude from a two's compliment negative
+        let n = u128::MAX - (n as u128) + 1;
+
+        Self::from_u128_sign(cx, Self::NEGATIVE, n)
+    }
+
+    /// Creates a `BigInt` from a [`u128`].
+    pub fn from_u128<'cx, C>(cx: &mut C, n: u128) -> Handle<'cx, Self>
+    where
+        C: Context<'cx>,
+    {
+        Self::from_u128_sign(cx, Self::POSITIVE, n)
+    }
+
+    /// Creates a `BigInt` from a signed magnitude (i.e., _not_ two's complement).
```

Comment:
This needs some more explanation of what endianness means precisely here, especially what happens if you're building on a big-endian device.